pydantic/core_utils.py

```diff
@@ -0,0 +1,177 @@
+"""
+The goal is to move this file's contents into pydantic_core proper.
```

Comment:
I put this stuff in regular pydantic for now to keep it simpler to get feedback, but I think all of this probably belongs in pydantic core if we find this useful.